docs/source/spec/aws-auth.rst

```diff
@@ -0,0 +1,125 @@
+.. _aws-authentication:
```

Comment:
Like protocols, I broke authentication traits into their own doc. Unlike protocols, I didn't give each trait it's own page. We could consider doing that if we perhaps add more extensive sigv4 documentation (for example, we may want to recreate the [sigv4 test suite](https://docs.aws.amazon.com/general/latest/gr/signature-v4-test-suite.html) to be Smithy protocol tests).